CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramTopComponent.java

```diff
@@ -637,7 +581,201 @@ protected void edit(final GraphWriteMethods graph, final PluginInteraction inter
 
         @Override
         public String getName() {
-            return "Histogram: Update State";
+            return "Histogram View: Update State";
+        }
+    }
+
+    /**
+     * Plugin to only select bins.
+     */
+    @PluginInfo(pluginType = PluginType.SELECTION, tags = {"SELECT"})
+    private class HistogramSelectOnlyBins extends SimpleEditPlugin {
+
+        private final int firstBin;
+        private final int lastBin;
+
+        public HistogramSelectOnlyBins(final int firstBin, final int lastBin) {
+            this.firstBin = firstBin;
+            this.lastBin = lastBin;
+        }
+
+        @Override
+        public String getName() {
+            return "Histogram View: Select Only Bins";
+        }
+
+        @Override
+        protected void edit(final GraphWriteMethods graph, final PluginInteraction interaction, final PluginParameters parameters) throws InterruptedException {
+            currentBinCollection.selectOnlyBins(graph, firstBin, lastBin);
+            if (selectedAttribute != Graph.NOT_FOUND) {
+                currentSelectedModificationCount = graph.getValueModificationCounter(selectedAttribute);
+            }
+        }
+    }
+
+    /**
+     * Plugin to select bins.
+     */
+    @PluginInfo(pluginType = PluginType.SELECTION, tags = {"SELECT"})
+    private class SelectBins extends SimpleEditPlugin {
+
+        private final int firstBin;
+        private final int lastBin;
+        private final boolean select;
+
+        public SelectBins(final int firstBin, final int lastBin, final boolean select) {
+            this.firstBin = firstBin;
+            this.lastBin = lastBin;
+            this.select = select;
+        }
+
+        @Override
+        public String getName() {
+            return "Histogram View: Select Bins";
+        }
+
+        @Override
+        protected void edit(final GraphWriteMethods graph, final PluginInteraction interaction, final PluginParameters parameters) throws InterruptedException {
+            currentBinCollection.selectBins(graph, firstBin, lastBin, select);
+            if (selectedAttribute != Graph.NOT_FOUND) {
+                currentSelectedModificationCount = graph.getValueModificationCounter(selectedAttribute);
+            }
+        }
+    }
+
+    /**
+     * Plugin to invert bins.
+     */
+    @PluginInfo(pluginType = PluginType.UPDATE, tags = {"MODIFY"})
```

Comment:
@sol695510 curios to know the reason you went with `MODIFY`. I was thinking `SELECTION` as according to the code it looks like the bins selected would change so the end result is just a selection change. I was however unable to find the "option" in the Histogram to trigger this.

Reply:
I'm looking at it again now and can't figure why I did PluginType.UPDATE and MODIFY, I've changed them to PluginType.SELECTION and SELECT. Same for the other two.